File: hun/prog2/java/penteki_gyakorlatok/week_13/pyutils_range/PyUtils.cs
using System.Collections.Generic;

namespace Example
{
    public class PyUtils
    {
        public static List<int> Range(int start, int end, int step)
        {
            var result = new List<int>();

            for (var i = start; i < end; i += step)
            {
                result.Add(i);
            }

            return result;
        }

        public static List<int> Range(int start, int end)
        {
            return PyUtils.Range(start, end, 1);
        }

        public static List<int> Range(int end)
        {
            return PyUtils.Range(0, end);
        }
    }
}

File: hun/prog2/java/penteki_gyakorlatok/week_04/ranger/PyUtils.java
import java.util.ArrayList;
import java.util.List;

public class PyUtils
{
    private PyUtils() {
        // nem példányosítható
    }

    public static List<Integer> range(int start, int end, int step)
    {
        List<Integer> result = new ArrayList<Integer>();

        for (int i = start; i < end; i += step) {
            result.add(i);
        }

        return result;
    }

    public static List<Integer> range(int start, int end)
    {
        return range(start, end, 1);
    }

    public static List<Integer> range(int end)
    {
        return range(0, end, 1);
    }
}
